### DISCUSSION TITLE
LongLLaMA: Focused Transformer: Contrastive Training for Context Scaling

### DISCUSSION THREAD

**Discussion by daboe01** [ggml-org/llama.cpp#2173] [Ideas]
https://www.marktechpost.com/2023/07/10/meet-longllama-a-large-language-model-capable-of-handling-long-contexts-of-256k-tokens/?amp
https://arxiv.org/abs/2307.03170
https://github.com/cstankonrad/long_llama

**Comment by ghost** [comment]
Hi,

It appears models with Focused Transformer(FoT) improve perplexity as context increases up to 64k, is that right?

**Comment by FNsi** [comment]
[4#](https://github.com/CStanKonrad/long_llama/issues/4)
